pkg/cloud/instance.go

```diff
@@ -198,14 +197,6 @@ func (c *client) GetOrCreateVMInstance(
 	setIfNotEmpty(csCluster.Spec.Account, p.SetAccount)
 	setIfNotEmpty(csCluster.Status.DomainID, p.SetDomainid)
 
-	// If this VM instance is a control plane, consider setting its IP.
-	_, isControlPlanceMachine := capiMachine.ObjectMeta.Labels["cluster.x-k8s.io/control-plane"]
-	if isControlPlanceMachine && zone.Network.Type == NetworkTypeShared {
-		// If the specified control plane endpoint is an IP address, specify the IP address of this VM instance.
-		if net.ParseIP(csCluster.Spec.ControlPlaneEndpoint.Host) != nil {
-			p.SetIpaddress(csCluster.Spec.ControlPlaneEndpoint.Host)
-		}
-	}
```

Comment:
Also would you mind adding more details to the PR description explaining the motivation for this change and what impact it would have? I think this might be a PR we refer back to in the future

Reply:
Updated the description. 
Even with single CP, I think it's better not to assign IP address to the CP because the kube-vip resolves the control plane endpoint IP address.